lib/circuit/CircuitStateStorage/ArgumentsCircuitStateStorage.ts

```diff
@@ -0,0 +1,22 @@
+import * as hash from 'object-hash';
+
+import { CircuitState } from '../CircuitState/CircuitState';
+import { CircuitStateFactory } from '../CircuitState/factory';
+import { CircuitStateStorage } from './CircuitStateStorage';
+
+export class ArgumentsCircuitStateStorage implements CircuitStateStorage {
+
+  private readonly argumentsStorage = new Map<string, CircuitState>();
```

Comment:
Will the elements from this map ever be deleted? Aren't we introducing a memory leak here?

Reply:
I added logic to remove keys from this map in last, commit I hope implementation is ok